services/ingestion/engine.go

```diff
@@ -206,6 +206,7 @@ func (e *Engine) processEvents(events *models.CadenceEvents) error {
 	}
 
 	e.collector.EVMHeightIndexed(events.Block().Height)
+	e.collector.CadenceHeightIndexed(events.CadenceHeight())
```

Comment:
I think we should change flow EVM GW to expect one evm block for one cadence block, now we kinda generalize the idea so there could be none blocks etc, I think this loose requirement can create more issues than it will solve, if we, one day change this mapping 1:1 we should update the gateway then. I don't think it's still worth generalizing at this point. So this PR can be accepted in the current shape but follow-up PR must be made to make ingesting more strict imho.